code/game/machinery/atmos_exporter.dm

```diff
@@ -0,0 +1,293 @@
+/obj/machinery/atmos_points
+	name = "atmospheric export computer"
+	desc = "used to monitor the galactic atmos markets"
+	icon = 'icons/obj/machines/engi_points.dmi'
+	icon_state = "atmos"
+	verb_say = "states"
+	density = 1
+	anchored = 1
+	var/market_cooldown = 0
+	var/export_target = ""
+	var/export_name = "------"
+	var/list/gas = list()
+	var/list/concentration = list()
+
+	// requirements of the canister we want to export
+	var/upressure = 0
+	var/lpressure = 0
+	var/utemp = 0
+	var/ltemp = 0
+	var/price = 200	
+	var/can_turf
+	// requirements for the economy canister
+	var/list/e_gas = list() 
+	var/economy_name = ""
+	var/e_pressure = 0
+	var/e_temp = 0
+	var/e_price = 200
+	var/e_concentration1 = 0
+	// requirements for the standard canister
+	var/standard_name = ""	
+	var/list/s_gas = list() 
+	var/s_pressure = 0
+	var/s_temp = 0
+	var/s_price = 600
+	var/s_concentration1 = 0
+	var/s_concentration2 = 0
+	// requirements for the premium canister
+	var/premium_name = ""
+	var/list/p_gas = list() // premium stats
+	var/p_pressure = 0
+	var/p_temp = 0
+	var/p_price = 1000
+	var/p_concentration1 = 0
+	var/p_concentration2 = 0
+	var/p_concentration3 = 0
+	var/list/quality = list()
+
+/obj/machinery/atmos_points_exporter
+	name = "atmospheric canister exporter"
+	desc = "used to export canisters for engineering points"
+	icon = 'icons/obj/machines/teleporter.dmi'
+	icon_state = "floorflush_c"
+	verb_say = "states"
+	density = 0
+	anchored = 1
+	
+/obj/machinery/atmos_points_exporter/Initialize()
+	..()
+	atmos_exporter_list += src
+
+/obj/machinery/atmos_points_exporter/Destroy()
+	..()
+	atmos_exporter_list -= src
+
+/obj/machinery/atmos_points_exporter/proc/export()
+	for(var/obj/machinery/portable_atmospherics/canister/C in get_turf(src))
+		animate(C, transform = matrix() - matrix(), alpha = 0, color = rgb(0, 0, 0), time = 10)
+		for(var/t in 1 to 5)
+			C.pixel_y--
+			sleep(2)
+			if(!C || QDELETED(C) || !src || QDELETED(src))
+				return
+		icon_state = "floorflush_a2"
+		qdel(C)
+			
+/obj/machinery/atmos_points_exporter/proc/ship()
+	icon_state = "floorflush_a"
+	playsound(src, 'sound/machines/Ding.ogg', 100, 1)
+	addtimer(CALLBACK(src, .proc/export), 20)
+
+
+/obj/machinery/atmos_points/Initialize()
+	..()
+	quality = list("Economy","Standard","Premium")
+	e_gas += "Initializing"
+	s_gas += "Initializing"
+	s_gas += "Initializing"
+	p_gas += "Initializing"
+	p_gas += "Initializing"
+	p_gas += "Initializing"
+
+/obj/machinery/atmos_points/interact(mob/user)
+	user.set_machine(src)
+	for(var/obj/machinery/atmos_points_exporter/AE in machines)
+		can_turf = get_turf(AE)
+	var/dat = text("Time until market update: [round((market_cooldown - world.time)/10)] seconds<br>")

```

Comment:
1. you're not even using the ```text()``` function correctly, so everywhere here you use it, just remove it (it's not doing anything and is unnecessary here)
2. define dat as a list, and at the end, just before you ```set_content()```. do dat.Join() to join the list dat back into one string (this is for performance)